docs/proposals/llm-d.md

```diff
@@ -81,6 +84,39 @@ Our current Northstar designs lay out the initial scope (join llm-d-contributors
 * [Prefix Cache Hierarchy](https://docs.google.com/document/d/1inTneLEZTv3rDEBB9KLOB9K6oMq8c3jkogARJqdt_58/edit)
 * [Variant Autoscaling over Hardware, Workload, and Traffic](https://docs.google.com/document/d/1inTneLEZTv3rDEBB9KLOB9K6oMq8c3jkogARJqdt_58/edit)
 
+### Components
+
+llm-d streamlines deployment and integration of the following components:
+
+1. An **inference scheduler** directing traffic to model servers
+    1. Integrates with the Kubernetes [inference gateway API project](https://gateway-api-inference-extension.sigs.k8s.io/) to have Kubernetes control plane API
+    2. Performing model routing and rollout, flow control, kv- and prefix- cache aware load balancing
+    3. Balances traffic to the optimal model server based on the request, workload type, and current load
+2. **vLLM model servers** deployed onto Kubernetes
+    1. In single host or multi-host (using [LeaderWorkerSets](https://lws.sigs.k8s.io/) and Ray as best practice) configurations
+    2. With native support for disaggregated serving and optional curated plugins for advanced features
+    3. Using project recommended defaults or highly customized user settings
+    4. May be deployed in multiple deployment **variants** (hardware, software, topology) that offer different performance tradeoffs
+3. vLLM default prefix caching and zero or more **prefix cache integrations**
+    1. At least one remote prefix cache option
+    2. At least one prefix cache option using the local SSD drives for each replica efficiently
+4. A **variant autoscaler** working with the inference scheduler and Kubernetes horizontal pod autoscaling
+    1. Can reassign prefill and decode roles between model server instances dynamically
+    2. Is aware of multiple deployment variants and their performance and can optimize across them
+    3. Can perform more advanced bucketization of traffic on latency or throughput objectives
+
+### APIs
+
+llm-d intends to drive the following APIs in our upstreams:
+
+1. vLLM public APIs to support inference scheduler-driven disaggregated serving
+2. vLLM internal APIs to support point-to-point disaggregated serving, remote prefix cache, and testing
```

Comment:
Also APIs to dynamically load models or LoRA adapters?

Reply:
LoRA Controller also happens to be one of the roadmap items in ModelService.